WalletWasabi/Blockchain/BlockFilters/IndexBuilderService.cs

```diff
@@ -77,8 +75,7 @@ public IndexBuilderService(IndexType indexType, IRPCClient rpc, BlockNotifier bl
 	private IRPCClient RpcClient { get; }
 	private BlockNotifier BlockNotifier { get; }
 	private string IndexFilePath { get; }
-	private List<FilterModel> Index { get; }
-	private AsyncLock IndexLock { get; }
+	private FilterLinkedList Index { get; }
```

Comment:
It is correct. That's what I did, I created a specialized persistent collection. Locking it is not necessary because it never changes, only during the creation of a new one (new head) you need to lock.